collect_app/src/main/java/org/odk/collect/android/utilities/FileUtils.java

```diff
@@ -400,9 +404,25 @@ public static void deleteAndReport(File file) {
         }
     }
 
+    public static String getFormFileName(File formXml) {
+        return getFormFileName(formXml.getName());
+    }
+
+    public static String getFormFileName(String formFilePath) {
+        return formFilePath.substring(0, formFilePath.lastIndexOf('.'));
+    }
+
     public static String constructMediaPath(String formFilePath) {
-        String pathNoExtension = formFilePath.substring(0, formFilePath.lastIndexOf('.'));
-        return pathNoExtension + "-media";
+        return getFormFileName(formFilePath) + "-media";
+    }
+
+    public static File getFormMediaDir(File formXml) {
+        final String formFileName = getFormFileName(formXml);
+        return new File(formXml.getParent(), formFileName + "-media");
```

Comment:
Perhaps the `-media` suffix should be a constant? Repeating that string literal seems like a bit of a smell.